### PR TITLE
Ability to have multiple speculate! blocks in the same module (fixes #7)

### LINKED ISSUE #7: Allow arbitrarily naming generated modules
I'd like to segment my tests into multiple sections of a file. For example, a structure would be defined, and then some tests related to this structure would appear in a `speculate!` block, and another structure would appear, etc. This would look like this:

```rust
pub struct Foo(...);

speculate! {
    it "can do things" {
        // Test Foo here.
    }
}

pub struct Bar(...);

speculate! {
    it "can do other things" {
        // Test Bar here.
    }
}
```

However, `speculate` always generate a module named `speculate`, which causes conflicts at the top-level.

The current workaround is to put every `speculate` macro in a individual, private modules, but it adds an unnecessary layer of nesting.

Instead, it would be nice if one of the following syntaxes would be added:

```rust
speculate! module_name { }
speculate! "module_name" { }
```

or if `speculate` appended, for example, the current line number to the created module name.

### PR DESCRIPTION
**Pull request by 6A**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -21,18 +21,12 @@ pub fn plugin_registrar(reg: &mut Registry) {
...
+fn expand_speculate(cx: &mut ExtCtxt, sp: Span, tokens: &[TokenTree]) -> Box<MacResult + 'static> {
+    let mod_name = format!("speculate #{}", sp.lo().0);
+    
     let mut parser = stream_to_parser(cx.parse_sess(), tokens.iter().cloned().collect());
-    let block = parser::parse(&mut parser);
+    let block = parser::parse(&mod_name, &mut parser);
     let item = block.generate(cx, None);
 
-    let module = quote_item!(cx,
-                             #[allow(non_snake_case)]
```
Comment (utkarshkukreti) [inline src/lib.rs]: We'll still need this or the compiler shows a warning for e.g. `describe "Node" {` (this throws dozens of warnings in https://github.com/utkarshkukreti/select.rs).

File: src/lib.rs [hunk 2]
```diff
@@ -21,18 +21,12 @@ pub fn plugin_registrar(reg: &mut Registry) {
...
-                             mod speculate {
-                                 #[allow(unused_imports)]
-                                 use super::*;
```
Comment (utkarshkukreti) [inline src/lib.rs]: I wonder why this isn't needed anymore.

Comment (6A) [reply src/lib.rs]: I tried a few things with top-level imports from `std` as well as some functoons, and `use super::*;` didn't change anything. I still pushed the changes that add `import super::*` everywhere.

**Comment by utkarshkukreti** [comment]
I don't completely understand the APIs you've had to use to make that import work but I'll merge since the tests are passing. Thanks! Let me know if you want a new release pushed to crates.io.

**Pull request merged**